SwiftyStoreKit/PaymentQueueController.swift

```diff
@@ -91,6 +94,26 @@ class PaymentQueueController: NSObject, SKPaymentTransactionObserver {
         paymentQueue.remove(self)
     }
 
+	var unhandledTransactions: [Purchase] {
+		guard let completeTransactions = completeTransactionsController.completeTransactions else {
+			print("SwiftyStoreKit.completeTransactions() should be called once when the app launches. Returning no results")
+			return []
+		}
+		let purchases = paymentQueue.transactions.flatMap { transaction -> Purchase? in
+			guard transaction.transactionState != .purchasing else {
+				return nil
+			}
+			return Purchase(
+				productId: transaction.payment.productIdentifier,
+				quantity: transaction.payment.quantity,
+				transaction: transaction,
+				originalTransaction: transaction.original,
+				needsFinishTransaction: !completeTransactions.atomically
```

Comment:
Yes, using `StoreKit` technically works but we'd not want to go that route since the code written for the fulfilment code can't be reused (as we're reusing fulfilment code in normal purchase flow as well as in handling of unfinished transactions as stated above. It has more than just `finishTransaction`: analytics, retry logic, delegate, etc..).